File: test/language/subtypes/java/subtypes/EnumSubtypeTest.java
package subtypes;

import static org.junit.jupiter.api.Assertions.*;
import org.junit.jupiter.api.Test;

import subtypes.enum_subtype.Color;
import subtypes.enum_subtype.CONST_BLACK;

public class EnumSubtypeTest
{
    @Test
    public void testSubtype()
    {
        assertEquals(Color.BLACK, CONST_BLACK.CONST_BLACK);
    }
}

File: test/extensions/language/subtypes/cpp/EnumSubtypeTest.cpp
#include <cstdio>
#include <fstream>
#include <string>

#include "gtest/gtest.h"
#include "subtypes/enum_subtype/CONST_BLACK.h"
#include "subtypes/enum_subtype/Color.h"

namespace subtypes
{
namespace enum_subtype
{

TEST(EnumSubtypeTest, testSubtype)
{
    ASSERT_EQ(Color::BLACK, CONST_BLACK);
}

} // namespace enum_subtype
} // namespace subtypes
